apps/ejabberd/src/foreign_event/mod_foreign_http.erl

```diff
@@ -28,9 +28,11 @@
 %%--------------------------------------------------------------------
 
 make_request(Request, OnResponse) ->
-    {Host, Path, Method, Headers, Body} = parse(Request),
-    Response = do_make_request(Host, Path, Method, Headers, Body),
-    respond(encode(Response), OnResponse),
+    spawn(fun() ->
```

Comment:
@arcusfelis the use case we have is that we need to make the requests asynchronously. The actual response of the HTTP request is to be published to MUC or PubSub.